impl/maven-di/src/main/java/org/apache/maven/di/impl/InjectorImpl.java

```diff
@@ -220,7 +220,13 @@ public <Q> Supplier<Q> doGetCompiledBinding(Dependency<Q> dep) {
         if (key.getRawType() == List.class) {
             Set<Binding<Object>> res2 = getBindings(key.getTypeParameter(0));
             if (res2 != null) {
-                List<Supplier<Object>> list = res2.stream().map(this::compile).collect(Collectors.toList());
+                // Sort bindings by priority (highest first) for deterministic ordering
+                List<Binding<Object>> sortedBindings = new ArrayList<>(res2);
+                Comparator<Binding<Object>> comparing = Comparator.comparing(Binding::getPriority);
+                sortedBindings.sort(comparing.reversed());
+
+                List<Supplier<Object>> list =
+                        sortedBindings.stream().map(this::compile).collect(Collectors.toList());
                 //noinspection unchecked
                 return () -> (Q) list(list, Supplier::get);
```

Comment:
considering this domain logic, might use OOP implementing `comparable` as its wont hurt nobody giving special attention and cohesion to core logic.

<img width="1862" alt="image" src="https://github.com/user-attachments/assets/416d7025-ea50-4bca-9d00-afa3b1fc00a2" />